CorePayments/src/main/java/com/paypal/android/corepayments/features/eligibility/EligibilityRequest.kt

```diff
@@ -0,0 +1,12 @@
+package com.paypal.android.corepayments.features.eligibility
+
+import com.paypal.android.corepayments.OrderIntent
+
+/**
+ * @property [intent] An order intent used in the request for eligibility.
+ * @property [currencyCode] A currency code used in the request for eligibility.
+ */
+data class EligibilityRequest(
+    val intent: OrderIntent,
```

Comment:
Do we want to call this `EligibilityIntent` to align with iOS?

Reply:
That's an interesting callout I wonder if iOS may want to rename to `OrderIntent`? The `intent` property has meaning outside of Eligibility–I believe it's tied to the Order domain.

Reply:
Yeh the Intent is widespread. It lives with the orderID from creation all the way through to it's final approval. Here's how our Orders API public docs describe it https://github.com/braintree/braintree_ios/pull/1370

Reply:
IMO just Intent is enough

Reply:
Are these the only intents (across any flow) that we will accept? If so agree we can go with generic `Intent` - I had assumed other requests would need different intents which was why it initially leaned `EligibilityIntent`. If that's not a concern lets do it!

Reply:
Yeah I also like `Intent` but there's already an Android `Intent` class. It may be better for merchants if we go with `OrderIntent` on Android to avoid a class name collision.